docs/r2dbc.md

```diff
@@ -0,0 +1,284 @@
+    .option(USER, "<DB_USER>")
+    .option(DATABASE, "<DATABASE_NAME}")
+    .option(HOST, "<CLOUD_SQL_CONNECTION_NAME>")
+    .build();
+
+ConnectionFactory connectionFactory = ConnectionFactories.get(options);
+ConnectionPoolConfiguration configuration = ConnectionPoolConfiguration
+    .builder(connectionFactory)
+    .build();
+
+ConnectionPool connectionPool = new ConnectionPool(configuration);
+```
+
+### IAM Authentication
+
+*Note:* This feature is currently only supported for Mysql and Postgres drivers.
+
+Connections using
+[IAM database authentication](https://cloud.google.com/sql/docs/postgres/iam-logins)
+are supported when connecting to Postgres instances.
+This feature is unsupported for other drivers. First, make sure to
+[configure your Cloud SQL Instance to allow IAM authentication](https://cloud.google.com/sql/docs/postgres/create-edit-iam-instances#configure-iam-db-instance)
+and
+[add an IAM database user](https://cloud.google.com/sql/docs/postgres/create-manage-iam-users#creating-a-database-user).
+Now, you can connect using user or service
+account credentials instead of a password.
+When setting up the connection, set the `ENABLE_IAM_AUTH` connection property to `true` and `user`
+to the email address associated with your IAM user.
+
+Note: a non-empty string value for the `password` property must be set. While this property will
+be ignored when connecting with the Cloud SQL Connector using IAM auth, leaving it empty will cause
+driver-level validations to fail.
+
+#### Mysql
+
+```java
+    // Set up ConnectionFactoryOptions
+ConnectionFactoryOptions options = ConnectionFactoryOptions.builder()
+    .option(DRIVER, "gcp")
+    .option(PROTOCOL, "mysql")
+    .option(PASSWORD, "password")
+    .option(USER, "mysql-iam-user@gmail.com")
+    .option(DATABASE, "my_db")
+    .option(HOST, "project:region:instance")
+    .option(ENABLE_IAM_AUTH, true)
+    .build();
+
+// Initialize connection pool
+ConnectionFactory connectionFactory = ConnectionFactories.get(options);
+ConnectionPoolConfiguration configuration = ConnectionPoolConfiguration
+    .builder(connectionFactory)
+    .build();
+
+this.connectionPool = new ConnectionPool(configuration);
+```
+
+
+#### Postgres
+
+```java
+// Set up ConnectionFactoryOptions
+ConnectionFactoryOptions options = ConnectionFactoryOptions.builder()
+    .option(DRIVER, "gcp")
+    .option(PROTOCOL, "postgresql")
+    .option(PASSWORD, "password")
+    .option(USER, "postgres-iam-user@gmail.com")
+    .option(DATABASE, "my_db")
+    .option(HOST, "project:region:instance")
+    .option(ENABLE_IAM_AUTH, true)
+    .build();
+
+// Initialize connection pool
+ConnectionFactory connectionFactory = ConnectionFactories.get(options);
+ConnectionPoolConfiguration configuration = ConnectionPoolConfiguration
+    .builder(connectionFactory)
+    .build();
+
+this.connectionPool = new ConnectionPool(configuration);
+```
+
+#### SQL Server
+
+Not Supported
+
+## Service Account Delegation
```

Comment:
should it be `Service Account Impersonation` ?

Reply:
Yes. Fixed.